collectors/test-results/jira-xray/src/main/java/com/capitalone/dashboard/core/client/testexecution/TestExecutionClientImpl.java

```diff
@@ -322,17 +322,24 @@ private TestCase createTestCase(TestExecution.Test test, TestRun testRun, Featur
                 testCase.setStatus(TestCaseStatus.Unknown);
             }
 
-             Set<String> tags = getStoryIds(feature.getIssueLinks());
-            // Temporarily commented for core project update
-            // testCase.setTags(tags);
             testCase.setTestSteps(this.getTestSteps(testRun));
+            List<Feature> tc = featureRepository.getStoryByNumber(test.getKey());
+            if (tc.size() > 0) {
+                for (Feature t : tc) {
+                    Set<String> tags = getStoryIds(t.getIssueLinks());
+                    testCase.setTags(tags);
+                }
+            }
+
         }
         return testCase;
     }
 
     private Set<String> getStoryIds(Collection<FeatureIssueLink> issueLinks) {
         Set<String> tags = new HashSet<>();
-        issueLinks.forEach(issueLink -> tags.add(issueLink.getTargetIssueKey()));
+        issueLinks.forEach(issueLink -> {if(issueLink.getIssueLinkType().equalsIgnoreCase("tests")){
```

Comment:
Can your format this better? The single line for part of this makes it hard to read.

Reply:
code formatted as suggested.